src/nvim/drawline.c

```diff
@@ -2110,21 +2106,34 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool nochange,
       }
       ptr++;
 
+      decor_attr = 0;
       if (extra_check) {
         bool no_plain_buffer = (wp->w_s->b_p_spo_flags & SPO_NPBUFFER) != 0;
         bool can_spell = !no_plain_buffer;
 
+        v = (ptr - line);
+        if (has_decor && v > 0) {
+          decor_attr = extmark_attr;
+
+          decor_conceal = decor_state.conceal;
+          if (decor_conceal && decor_state.conceal_char) {
+            decor_conceal = 2;  // really??
+          }
+
+          can_spell = TRISTATE_TO_BOOL(decor_state.spell, can_spell);
+        }
+
         // Get syntax attribute, unless still at the start of the line
```

Comment:
Keeping this comment on top (and edited to mention extmark instead of syntax) like #23726 did might be preferred?

Reply:
done